### PR TITLE
Make 3PID binding tests use /account/3pid/bind ala MSC2290

### PR DESCRIPTION
**Pull request by anoadragon453**
Synapse PR: https://github.com/matrix-org/synapse/pull/6043

Changes the 3PID binding tests to use the new unstable `/account/3pid/bind` API endpoint, defined in [MSC2290](https://github.com/matrix-org/matrix-doc/pull/2290).

### CODE REVIEW

**Comment by anoadragon453** [comment]
The CI here is going to fail as it's using Synapse develop not Synapse `anoa/msc2290` :/

**Comment by anoadragon453** [comment]
@richvdh There are some merge conflicts introduced here due to `add_email_to_user` replacing the `/requestToken` calls here. Can you sort out what should be done?

**Comment by richvdh** [comment]
done

**Comment by richvdh** [comment]
@anoadragon453 I think I now need you to rreview this one, ironically...

File: tests/12login/01threepid-and-password.pl [hunk 1]
```diff
@@ -120,24 +120,11 @@ =head2 add_email_for_user
...
-=item bind => SCALAR
-
-If truthy, we will ask the server to ask the IS to bind the email
-address. False by default.
-
-=back
-
 =cut
 
 sub add_email_for_user {
```
Comment (anoadragon453) [inline tests/12login/01threepid-and-password.pl]: If `bind_email_for_user` is using the new bind API, should this method use the new add API?

Comment (richvdh) [reply tests/12login/01threepid-and-password.pl]: I guess... we can do that later though?

Comment (anoadragon453) [reply tests/12login/01threepid-and-password.pl]: True

**Comment by richvdh** [comment]
this is now ready to merge, pending https://github.com/matrix-org/synapse/pull/6043.

**Pull request merged**